### DISCUSSION TITLE
Merge dialog: new behaviour in 3.38.0 is worse than before

### DISCUSSION THREAD

**Discussion by beanzie** [mifi/lossless-cut#832] [Ideas]
in 3.37.0, the user can control the order in which multiple files appear in the merge dialog by changing the order in which the files are selected using their file explorer/finder etc (immediately before dragging the files into Losslesscut)

in 3.38.0, the order will automatically change the sorting of the files to be merged to sort it by filename. I would argue this is actually regressive as it removes the control described above, and introduces eccentricities to do with filename conventions.

example: you want to merge 20 files that are called "a1.mp4, a2.mp4, a3.mp4 ... a20.mp4". the files are named in sequential order.

in 3.37.0 you can just select the files starting with a20.mp4, finishing with a1.mp4, and drag and drop into Losslesscut. When done this way, the files will appear in the following order:

- a1.mp4
- a2.mp4
- a3.mp4
- ... a20.mp4

in other words it is fairly simple to get the desired order immediately upon dropping the files into Losslesscut. you need to select them in your file manager a specific way, but that's more of an operating system issue than a Losslesscut issue.

now in 3.38.0, if you follow the same steps above the order instead is:

- a1.mp4
- a10.mp4
- a11.mp4
- ... a19.mp4
- a2.mp4
- a20.mp4
- a3.mp4
- a4.mp4
- ... a9.mp4

I'll continue using 3.37.0 for now, but would suggest rolling back to the previous implementation or coming up with a new solution because the current behaviour's intent is good but in practice it's worse than it was before

**Comment by mifi** [comment]
Thanks for reporting. Which operating system are you on?

The reason why I implemented this change is that there was a request to get the files sorted, because by default they came out in a random order when drag-dropped, which seemed reasonable.

> when you drop multiple videos to merge, they are sorted almost randomly, even when my files are sorted by name. So when you drag 5+ files, instead of easily dragging and press enter, you must order them first. I dragged 15 files to merge and it took some time to order those files. Thanks for your time.
...from https://www.youtube.com/watch?v=pYHMxXy05Jg&lc=UgyLmiq5MRx5lWe_wdx4AaABAg

So I think the correct solution then would be to have a button to press to sort the list by name, then everyone would be happy.

**Comment by beanzie** [comment]
Thanks for your reply. I'm using the Windows 11 Developer Preview. I think the original reported issue could actually just be solved if the user experimented with different methods to drop the files into Losslesscut. Possibly they are making slight alterations to their method (like having a different segment clip highlighted when they click the left mouse button to drag and drop) and interpreting the different output as randomness. I'm vocal about this issue because due to hardware limitations I merge a lot of clips and in 3.37.0, I can just click Merge immediately because I can trust that the file order will automatically import the same way, every single time, based solely on the order in which I dropped it in.

Even if your files aren't named sequentially, proper use of the file manager can make the process very efficient and this should trump any automatic sorting done by Losslesscut. Say you want to merge the last 10 clips you edited regardless of filename, this case you can change the sorting method to "Last Modified" or "Created Date" etc in your file manager, drag and drop the files into Losslesscut in the order you want and again, have the files appear in the perfect order for your use case. Again this is a use case that 3.37.0 handles perfectly, and 3.38.0 does not.

I prefer the older behaviour personally, and I think ultimately if it came down to one method or the other then the method from 3.37.0 is clearly superior for a wider variety of use cases. But if you can provide both options or more powerful features for sorting files in general - that's awesome! Thanks for your development on this great app :)

**Reply by mifi** [reply]
I understand. I wasn't aware that finder/explorer allows customizing order of drag-dropped file. I have reverted the behavior and added a sort button in the merge dialog, so it will be out in the next version

**Reply by beanzie** [reply]
The merge dialog in 3.39.0 is much improved. thanks so much!
